src/server.cc

```diff
@@ -174,6 +174,7 @@ Status Server::Start() {
 
 void Server::Stop() {
   stop_ = true;
+  Lua::DestroyState(lua_);
```

Comment:
Can move this line to the server destructor and delete the server after the server threads join.